src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -948,10 +948,23 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             modifiers |= ControlKeyStates::EnhancedKey;
         }
 
-        const auto handled = _core.SendCharEvent(ch, scanCode, modifiers);
+        // Broadcast the character to all listeners
+        auto charSentArgs = winrt::make<CharSentEventArgs>(ch, scanCode, modifiers);
+        _CharSentHandlers(*this, charSentArgs);

```

Comment:
one last thing I'd be way of is the object cost. Every character, every key, is getting another heap allocated object that is bubbled up to somebody who is listening. It's not likely to be a bottleneck -- human typing speed is limited, after all -- but it's worth considering.